editor/gui/scene_tree_editor.cpp

```diff
@@ -1527,12 +1539,24 @@ void SceneTreeDialog::_filter_changed(const String &p_filter) {
 	tree->set_filter(p_filter);
 }
 
+void SceneTreeDialog::_show_all_nodes_changed(bool p_button_pressed) {
+	tree->set_show_all_nodes(p_button_pressed);
+}
+
 void SceneTreeDialog::_bind_methods() {
 	ClassDB::bind_method("_cancel", &SceneTreeDialog::_cancel);
 
 	ADD_SIGNAL(MethodInfo("selected", PropertyInfo(Variant::NODE_PATH, "path")));
 }
 
+void SceneTreeDialog::add_show_all_nodes_filter() {
+	show_all_nodes = memnew(CheckButton);
+	show_all_nodes->set_text(TTR("Show All Nodes"));
+	show_all_nodes->connect("toggled", callable_mp(this, &SceneTreeDialog::_show_all_nodes_changed));
+	show_all_nodes->set_h_size_flags(Control::SIZE_SHRINK_BEGIN);
+	filter->add_sibling(show_all_nodes);
+}
+
```

Comment:
I don't think it makes sense for this function to be split off.

Reply:
What's the ideal solution? I can't put this code in the constructor because, if I've seen correctly, the SceneTreeDialog is used in other cases without a type filter (valid_types), so the show_all_nodes CheckButton wouldn't make sense in these cases.

Do you think I should extract all the code relating to the new CheckButton and put it in editor_properties.cpp?

Reply:
Just copying this code in the single place the function is used.

Reply:
Should be good now